docs/loaders.md

```diff
@@ -126,42 +126,38 @@ Like with any other file, these files from generated archives are live in previe
 
 ## Routing
 
-Data loaders live in the source root (typically `docs`) alongside your other source files. When a file is referenced from JavaScript via `FileAttachment`, if the file does not exist, Observable Framework will look for a file of the same name with a double extension to see if there is a corresponding data loader. The following second extensions are checked, in order, with the corresponding language and interpreter:
+Data loaders live in the source root (typically `docs`) alongside your other source files. When a file is referenced from JavaScript via `FileAttachment`, if the file does not exist, Observable Framework will look for a file of the same name with a double extension to see if there is a corresponding data loader. By default, the following second extensions are checked with the corresponding language and interpreter:
 
 - `.js` - JavaScript (`node`)
 - `.ts` - TypeScript (`tsx`)
+- `.sh` - shell script (`sh`)
+- `.exe` - arbitrary executable
+- `.go` - Go (`go run`)
+- `.java` — Java (`java`; requires Java 11+ and [single-file programs](https://openjdk.org/jeps/330))
+- `.jl` - Julia (`julia`)
+- `.php` - PHP (`php`)
 - `.py` - Python (`python3`)
 - `.R` - R (`Rscript`)
 - `.rs` - Rust (`rust-script`)
```

Comment:
Changing the order of precedence would be non-backwards-compatible, and I don’t want to call this Framework 2.0. Can we restore the original order?

Reply:
I wanted to reorder alphabetically as this list is going to build up and then look messy. The order of precedence is not something we should design for: it would not be a great idea for a user to have two competing data loaders for a file, and to rely on this order to pick up the chosen data loaders. (It's also not something that the user can change with the **interpreters** option.)

You're right however that this is what the code currently does, and that changing the order will break someone's code. I'm thinking now that this should be addressed by erroring — instead of picking the first match from an arbitrary list. But if we don't error, then yes we should keep a backwards compatible order.

Reply:
Also consider the following scenario:

A user wants to write a script in a language that is not supported (say, fortran). They write their data loader as `data.csv.f77`, then realize it doesn't work, and write `data.csv.sh` which sets up a few variables and execs `data.csv.f77`.

A week later a new version of Framework is out and supports the `f77` extension.